paper/paper.Rmd

```diff
@@ -0,0 +1,308 @@
+while the latter is typically larger and provides information about the range individual observations might take (e.g., _around_ the expectation $E[X]$).
+Moreover, for generalized linear models (GLMs), predictions can be made on the **response scale** (`predict = "response"`) or the **link scale** (`predict = "link"`).
+This corresponds for instance to predictions in terms of probability (response scale) or log odds (link scale) for logistic regression models.
+
+These different types of estimates can be obtained for observations in the original dataset,
+which is useful to assess the model's goodness-of-fit,
+or for new data (typically a "data grid"), which is useful for visualization.
+
+For convenience, the `modelbased` package includes four related functions, that mostly differ in their default arguments for `data` and `predict`:
+
+- `estimate_prediction()`: original data, prediction intervals.
+- `estimate_expectation()`: original data, confidence intervals.
+- `estimate_relation()`: data grid, predictions on the response scale.
+- `estimate_link()`: data grid, predictions on the link scale.
+
+*Note:* if the defaults are changed, then these functions can become redundant.
+For instance, `estimate_relation(..., predict = "link")` will be equivalent to `estimate_link(...)`.
+
+Which function to use depends on the model and the research question at hand (visualizing effects, generating predictions, etc.).
+
+
+## Marginal means and effects
+
+The concept of "marginal" in statistical modeling typically refers to the analysis of the effect of one or more "focus" variables while all other variables are held constant at specific values
+(e.g., their empirical or theoretical average or reference value).
+This in turn is convenient for understanding the effect of a variable in a complex model, where multiple variables interact with each other.
+
+The `modelbased` package simplifies the extraction of these quantities, providing a clear interface to understand how different predictors interact with outcomes in various scenarios.
+
+- `estimate_means()`: computes **Marginal Means**, i.e., the average predictions for each level of a categorical predictor, averaged across all levels of other predictors.
+- `estimate_contrasts()`: computes **Marginal Contrasts**, i.e., the comparison of marginal means at different levels of a factor to assess differences or effects.
+- `estimate_slopes()`: computes **Marginal Effects**, i.e., the change in the response variable for an infinitesimal change in a predictor, averaged across all levels of other predictors. They are essentially the partial derivatives of the response with respect to each predictor, useful for continuous predictors, and describe the average strength of an effect.
+
+
+**[TODO: details about types of marginalization]**
+
+
+## Technical details
```

Comment:
Maybe we can shorten this section a bit? The paper becomes quite long, and I'm not sure if the technical details of the _backend_ packages are helpful to understand what **modelbased** is and how it works, and what's the idea behind our package?

Reply:
yes we can streamline a bit, although in general I like the idea of having slightly more substance in that paper compared to some of our other ones (that whole domain deserves it and it will be useful as a reference to readers)

Reply:
Ok agreed